ui/components/app/assets/account-group-balance-change/account-group-balance-change.tsx

```diff
@@ -0,0 +1,147 @@
+  currentTotalInUserCurrency: number;
+  previousTotalInUserCurrency: number;
+  amountChangeInUserCurrency: number;
+  percentChange: number;
+  userCurrency: string;
+};
+
+export const AccountGroupBalanceChange: React.FC<
+  AccountGroupBalanceChangeProps
+> = ({ period }) => {
+  const isMultichainAccountsState2Enabled = useSelector(
+    getIsMultichainAccountsState2Enabled,
+  ) as boolean;
+  const fiatCurrency = useSelector(getCurrentCurrency) as string;
+  const locale = useSelector(getIntlLocale) as string;
+  const { privacyMode } = useSelector(getPreferences);
+
+  // Memoized selector for the specified period
+  const changeSelector = useMemo(
+    () => selectBalanceChangeBySelectedAccountGroup(period),
+    [period],
+  );
+
+  // Get the data
+  const portfolioChange = useSelector(
+    isMultichainAccountsState2Enabled ? changeSelector : () => null,
+  ) as AggregatedChange | null;
+  const portfolioPercent = portfolioChange?.percentChange ?? 0;
+
+  // Early return if feature flag is off
+  if (!isMultichainAccountsState2Enabled) {
+    return null;
+  }
+
+  let color = TextColor.textDefault;
+  if (privacyMode) {
+    color = TextColor.textAlternative;
+  } else if (
+    portfolioChange &&
+    isValidAmount(portfolioChange.amountChangeInUserCurrency)
+  ) {
+    const amt = Number(portfolioChange.amountChangeInUserCurrency);
+    if (amt > 0) {
+      color = TextColor.successDefault;
+    } else if (amt < 0) {
+      color = TextColor.errorDefault;
+    }
+  } else if (isValidAmount(portfolioPercent)) {
+    const pct = Number(portfolioPercent);
+    if (pct > 0) {
+      color = TextColor.successDefault;
+    } else if (pct < 0) {
+      color = TextColor.errorDefault;
+    }
+  }
+
+  const percentNumber =
+    typeof portfolioPercent === 'number' ? portfolioPercent : 0;
+  const signPrefix = percentNumber >= 0 ? '+' : '-';
+  const absPercent = Math.abs(percentNumber);
+  const displayAbsPercent =
+    absPercent !== 0 && absPercent < 0.01 ? 0.01 : absPercent;
+  const localizedPercent = Intl.NumberFormat(locale, {
+    style: 'percent',
+    maximumFractionDigits: 2,
+    minimumFractionDigits: 2,
+  }).format(displayAbsPercent / 100);
+  const formattedPercentWithParens = `(${signPrefix}${localizedPercent})`;
+
+  let formattedAmount = '';
+  if (
+    portfolioChange &&
+    isValidAmount(portfolioChange.amountChangeInUserCurrency)
+  ) {
+    const amt = Number(portfolioChange.amountChangeInUserCurrency);
+    const signPrefixAmount = amt >= 0 ? '+' : '';
+    try {
+      formattedAmount = `${signPrefixAmount}${Intl.NumberFormat(locale, {
+        style: 'currency',
+        currency: fiatCurrency,
+        minimumFractionDigits: 2,
+        maximumFractionDigits: 2,
+      }).format(amt)} `;
+    } catch {
+      formattedAmount = `${signPrefixAmount}${Intl.NumberFormat(locale, {
+        style: 'decimal',
+        minimumFractionDigits: 2,
+        maximumFractionDigits: 2,
+      }).format(amt)} `;
+    }
+  }
```

Comment:
Is this not something that already exists as a util?
I would prefer moving this into a utility.

Reply:
We can test in isolation and not tied to components/state.